Sources/StreamChat/Database/DTOs/ChannelDTO.swift

```diff
@@ -124,6 +128,10 @@ extension NSManagedObjectContext {
         dto.lastMessageAt = payload.lastMessageAt
         dto.memberCount = Int64(clamping: payload.memberCount)
 
+        if let oldestMessageAt = dto.messages.map(\.createdAt).min() {
+            dto.oldestMessageAt = oldestMessageAt
+        }
```

Comment:
We should override only if `dto.oldestMessageAt < oldestMessageAt`. This would be good to test, too, btw.

Reply:
Why is that check necessary? 🤔